### PR TITLE
refactor encryptionkey resource for clusterapi controller

### PR DESCRIPTION
**Pull request by xh3b4sd**
The resource is not yet wired as we want to move all resources over first. I dropped the tests for now as they where too complicated and cumbersome to migrate.

### CODE REVIEW

File: service/controller/clusterapi/v18/key/cluster.go [hunk 1]
```diff
@@ -57,6 +57,12 @@ func ClusterName(cluster v1alpha1.Cluster) string {
 	return clusterProviderSpec(cluster).ClusterName
 }
 
+// EncryptionKeySecretName generates name for a Kubernetes secret based on
+// information in given v1alpha1.ClusterGuestConfig.
+func EncryptionKeySecretName(cr v1alpha1.Cluster) string {
```
Comment (kopiczko) [inline service/controller/clusterapi/v18/key/cluster.go]: Do we need to have that in the key package? It looks like it's used only by the encryptionkey resource.

File: service/controller/clusterapi/v18/resources/encryptionkey/current.go [hunk 2]
```diff
@@ -2,42 +2,36 @@ package encryptionkey
...
 
-	objectMeta, err := r.toClusterObjectMetaFunc(obj)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	secretName := key.EncryptionKeySecretName(clusterGuestConfig)
+	var secret *corev1.Secret
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding secret %#q", key.EncryptionKeySecretName(cr)))
```
Comment (kopiczko) [inline service/controller/clusterapi/v18/resources/encryptionkey/current.go]: I would also like to log the namespace.

File: service/controller/clusterapi/v18/key/cluster.go (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v18/key/cluster.go]: I like that. Moved to resource as private key function.

File: service/controller/clusterapi/v18/resources/encryptionkey/current.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v18/resources/encryptionkey/current.go]: sorted 🎉

**Comment by xh3b4sd** [comment]
I went with the state getter approach. Good shout. Thank you.

**Comment by xh3b4sd** [comment]
That is super nice. A thousand lines removed and a hundred added. ❤️

**Pull request merged**